src/googleError.ts

```diff
@@ -65,7 +65,18 @@ export class GoogleError extends Error {
     );
     // Map Http Status Code to gRPC Status Code
     if (json['error']['code']) {
-      error.code = rpcCodeFromHttpStatusCode(json['error']['code']);
+      const rpcCode = rpcCodeFromHttpStatusCode(json['error']['code']);
+      if (
+        !json['error']['status'] ||
+        Status[rpcCode] !== json['error']['status']
+      ) {
+        console.warn(
```

Comment:
`console.warn` goes to user. If they don't have anything to do with this, we must not spam their console. What can we do in this case? Can we still set `err` to something that makes sense?

Reply:
A user can't do anything if mismatch. HTTP server should do provide grpc-matched status.
https://cloud.google.com/apis/design/errors#error_mapping

In rare case, if http doesn't match grpc status, we have two options:
1. Use http code associated grpc code to find correct `status`.  This option would hide the mismatch information.
2. Create new error message.  But we have to decide which code and status need to use. code and status has to be [grpc status](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). This option changed the HTTP code. The way to make it clear, is to add more details in message.

My opinion is option 2.  Given detailed message include the http code and matched grpc code/status to explain the mismatch error, assign new error with code `2` status `UNKNOWN`. 

`UNKNOWN` description: Unknown error. For example, this error may be returned when a Status value received from another address space belongs to an error space that is not known in this address space. Also errors raised by APIs that do not return enough error information may be converted to this error.

`INTERNAL` is not the first choice because it reserve for the serious errors.